internal/auth0/organization/resource_connections.go

```diff
@@ -0,0 +1,230 @@
+								"automatically granted membership in the organization. When false, users must be " +
+								"granted membership in the organization before logging in with this connection.",
+						},
+					},
+				},
+				Required:    true,
+				Description: "IDs of the connections that are enabled for the organization.",
+			},
+		},
+		CreateContext: createOrganizationConnections,
+		ReadContext:   readOrganizationConnections,
+		UpdateContext: updateOrganizationConnections,
+		DeleteContext: deleteOrganizationConnections,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Description: "With this resource, you can manage enabled connections on an organization.",
+	}
+}
+
+func createOrganizationConnections(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	mutex := meta.(*config.Config).GetMutex()
+
+	organizationID := data.Get("organization_id").(string)
+
+	mutex.Lock(organizationID)
+	defer mutex.Unlock(organizationID)
+
+	alreadyEnabledConnections, err := api.Organization.Connections(organizationID)
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	data.SetId(organizationID)
+
+	connectionsToAdd := expandOrganizationConnections(data.GetRawConfig().GetAttr("enabled_connections"))
+
+	if diagnostics := guardAgainstErasingUnwantedConnections(
+		organizationID,
+		alreadyEnabledConnections.OrganizationConnections,
+		connectionsToAdd,
+	); diagnostics.HasError() {
+		data.SetId("")
+		return diagnostics
+	}
+
+	var result *multierror.Error
+	for _, connection := range connectionsToAdd {
+		if err := api.Organization.AddConnection(organizationID, connection); err != nil {
+			result = multierror.Append(result, err)
+		}
+	}
+
+	if result.ErrorOrNil() != nil {
+		return diag.FromErr(result.ErrorOrNil())
+	}
+
+	return readOrganizationConnections(ctx, data, meta)
+}
+
+func readOrganizationConnections(_ context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+
+	connections, err := api.Organization.Connections(data.Id())
+	if err != nil {
+		if mErr, ok := err.(management.Error); ok && mErr.Status() == http.StatusNotFound {
+			data.SetId("")
+			return nil
+		}
+
+		return diag.FromErr(err)
+	}
+
+	result := multierror.Append(
+		data.Set("organization_id", data.Id()),
+		data.Set("enabled_connections", flattenOrganizationConnections(connections.OrganizationConnections)),
+	)
+
+	return diag.FromErr(result.ErrorOrNil())
+}
+
+func updateOrganizationConnections(ctx context.Context, data *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	api := meta.(*config.Config).GetAPI()
+	mutex := meta.(*config.Config).GetMutex()
+	mutex.Lock(data.Id())
+	defer mutex.Unlock(data.Id())
```

Comment:
Just a Go sanity check, if the ID changes before the deferred function is called, the original ID value is still used, correct? Because the ID gets reassigned to empty string potentially.

Reply:
Good catch, the ID will become an empty string.

Reply:
Updated: [`6b762c5` (#610)](https://github.com/auth0/terraform-provider-auth0/pull/610/commits/6b762c5214791308f50896b865127585e94b135d)